src/app/clusters/resource-monitoring-server/resource-monitoring-server.h

```diff
@@ -0,0 +1,161 @@
+
+    // Attribute setters
+    chip::Protocols::InteractionModel::Status UpdateCondition(uint8_t aNewCondition);
+    chip::Protocols::InteractionModel::Status UpdateChangeIndication(ChangeIndicationEnum aNewChangeIndication);
+    chip::Protocols::InteractionModel::Status UpdateInPlaceIndicator(bool aNewInPlaceIndicator);
+    chip::Protocols::InteractionModel::Status UpdateLastChangedTime(DataModel::Nullable<uint32_t> aNewLastChangedTime);
+
+    // Attribute getters
+    uint8_t GetCondition() const;
+    ChangeIndicationEnum GetChangeIndication() const;
+    DegradationDirectionEnum GetDegradationDirection() const;
+    bool GetInPlaceIndicator() const;
+    DataModel::Nullable<uint32_t> GetLastChangedTime() const;
+
+private:
+    EndpointId mEndpointId{};
+    ClusterId mClusterId{};
+
+    // attribute Data Store
+    chip::Percent mCondition                       = 100;
+    DegradationDirectionEnum mDegradationDirection = DegradationDirectionEnum::kDown;
+    ChangeIndicationEnum mChangeIndication         = ChangeIndicationEnum::kOk;
+    bool mInPlaceIndicator                         = true;
+    DataModel::Nullable<uint32_t> mLastChangedTime;
+
+    uint32_t mFeatureMap;
+
+    bool mResetConditionCommandSupported = false;
+
+    /**
+     * This checks if the clusters instance is a valid ResourceMonitoring cluster based on the AliasedClusters list.
+     * @return true     if the cluster is a valid ResourceMonitoring cluster.
+     */
+    bool IsValidAliasCluster() const;
+
+    /**
+     * Internal method to handle the ResetCondition command.
+     */
+    void HandleResetCondition(HandlerContext & ctx,
+                              const ResourceMonitoring::Commands::ResetCondition::DecodableType & commandData);
+
+public:
+    /**
+     * Creates a resource monitoring cluster instance. The Init() method needs to be called for this instance to be registered and
+     * called by the interaction model at the appropriate times.
+     * @param aEndpointId   The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId    The ID of the ResourceMonitoring aliased cluster to be instantiated.
+     * @param aDelegate     A pointer to a delegate that will handle application layer logic.
+     */
+    Instance(EndpointId aEndpointId, ClusterId aClusterId, uint32_t aFeatureMap,
+             ResourceMonitoring::Attributes::DegradationDirection::TypeInfo::Type aDegradationDirection,
+             bool aResetConditionCommandSupported) :
+        CommandHandlerInterface(Optional<EndpointId>(aEndpointId), aClusterId),
+        AttributeAccessInterface(Optional<EndpointId>(aEndpointId), aClusterId), mDegradationDirection(aDegradationDirection),
+        mResetConditionCommandSupported(aResetConditionCommandSupported)
+    {
+        mEndpointId = aEndpointId;
+        mClusterId  = aClusterId;
+        mFeatureMap = aFeatureMap;
```

Comment:
Why the mix of initializers (mResetConditionCommandSupported) and assignments?  Just use initializers for all of them, please.

Reply:
done